app/search/gleaner.py

```diff
@@ -57,12 +57,16 @@ def build_query(text_query="", filter_query="", page_number=1):
                     FILTER(ISLITERAL(?identifier)) .
                 }}
                 OPTIONAL {{
-                    ?s schema:creator/schema:name ?author .
+                    
+                    {{ ?s schema:creator/schema:name ?author . }} UNION {{
```

Comment:
oh cool, do some of them have the authors on the datacatalog too? Nice.

Reply:
ah, ok -  that's the case that the stuff after UNION here will handle - it's grabbing the author off the DataCatalog level, like had to happen for keywords in CCHDO datasets. Take https://cchdo.ucsd.edu/cruise/06MT19960613#bottle for example - its metadata is the following, and you can see keywords defined at the DataCatalog level instead of the individual DataSet level:

`
{
    "@context":
    {
        "@vocab": "http://schema.org/",
        "geolink": "http://schema.geolink.org/1.0/base/main#"
    },
    "@id": "https://cchdo.ucsd.edu/cruise/06MT19960613",
    "@type": "DataCatalog",
    "dataset":
    [
        {
            "@id": "https://cchdo.ucsd.edu/cruise/06MT19960613#bottle",
            "@type": "Dataset",
            "description": "Bottle data from the hydrographic cruise 06MT19960613",
            "distribution":
            [
                {
                    "@type": "DataDownload",
                    "contentSize": "128854 bytes",
                    "contentUrl": "https://cchdo.ucsd.edu/data/10418/06MT19960613_hy1.csv",
                    "encodingFormat": "text/csv",
                    "name": "exchange"
                }
            ],
            "isAccessibleForFree": true,
            "name": "DE Hydrographic Cruise 06MT19960613: Bottle Dataset",
            "provider":
            {
                "@id": "https://cchdo.ucsd.edu"
            },
            "publisher":
            {
                "@id": "https://cchdo.ucsd.edu",
                "@type": "Organization",
                "name": "CCHDO",
                "url": "https://cchdo.ucsd.edu"
            },
            "spatialCoverage":
            {
                "@id": "https://cchdo.ucsd.edu/cruise/06MT19960613#spatialCoverage",
                "@type": "Place",
                "geo":
                [
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 1",
                        "latitude": 41.92,
                        "longitude": -34.08
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 2",
                        "latitude": 33.45,
                        "longitude": -22.16
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 3",
                        "latitude": 29.26,
                        "longitude": -15.51
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 4",
                        "latitude": 33.01,
                        "longitude": -22.02
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 5",
                        "latitude": 35.5,
                        "longitude": -21.33
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 6",
                        "latitude": 37.83,
                        "longitude": -20.67
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 7",
                        "latitude": 39.79,
                        "longitude": -20.06
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 8",
                        "latitude": 43.07,
                        "longitude": -19.83
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 9",
                        "latitude": 45.34,
                        "longitude": -19.83
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 10",
                        "latitude": 47.51,
                        "longitude": -19.78
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 11",
                        "latitude": 49.9,
                        "longitude": -20.24
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 12",
                        "latitude": 52.29,
                        "longitude": -20.69
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 13",
                        "latitude": 54.64,
                        "longitude": -21.16
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 14",
                        "latitude": 54.64,
                        "longitude": -21.15
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 15",
                        "latitude": 57.09,
                        "longitude": -20.62
                    },
                    {
                        "@type": "GeoCoordinates",
                        "description": "Profile 16",
                        "latitude": 59.48,
                        "longitude": -20.02
                    }
                ]
            },
            "temporalCoverage": "1996-06-13/1996-07-14"
        }
    ],
    "description": "Data from the hydrographic cruise 06MT19960613",
    "keywords":
    [
        "METEOR",
        "bottle",
        "Kremling",
        "CARINA",
        "atlantic",
        "06MT19960613"
    ],
    "name": "DE Hydrographic Cruise 06MT19960613"
}
`
 I don't know where their author information comes from (they have a Chief Scientist listed on the landing page), but it seems plausible to me that someone might decide to define author information in this way too.